clusterloader2/framework/util/kubeconfig.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+func restclientConfig(path string) (*clientcmdapi.Config, error) {
+	c, err := clientcmd.LoadFromFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("error loading KubeConfig: %v", err)
+	}
+	return c, nil
+}
+
+// LoadConfig loads kubeconfig from given location.
+func LoadConfig(path string) (*restclient.Config, error) {
+	c, err := restclientConfig(path)
+	if err != nil {
+		return restclient.InClusterConfig()
+	}
```

Comment:
I'm pretty sure we will need to tune some stuff.
Things like:
- QPS limits
- use protobufs as content-type

BTW - I also think we may want to use multiple clients (though that is less clear for now).